### DISCUSSION TITLE
select linter

### DISCUSSION THREAD

**Discussion by rootsmusic** [microsoft/vscode-pylint#441] [Q&A] [answered]
In VS Code 1.84.0 on Windows, I installed and enabled several extensions including [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python) and [Microsoft's linters](https://code.visualstudio.com/docs/python/linting#_choose-a-linter).  I've opened a .py file, and I don't see any error.  In my command palette, why can't I find "Python: Select Linter" ("Restart Server" commands are available for three linters)?  Thanks.

**Comment by karthiknadig** [comment]
@rootsmusic you can open Output > Pylint to see if there was any issue running linters. 

There is no select linter because once you install the Pylint extension, the linter is active. You don't need to turn it on.

**Reply by karthiknadig** [reply]
you should see something like this:
![image](https://github.com/microsoft/vscode-pylint/assets/3840081/ab3c515f-72c8-4e27-8b03-2fea86bed5e7)

if you are not, you might need to looks ate developer tools logs (Help > toggle Developer tools)

**Reply by karthiknadig** [reply]
The only other thing i can thing of is that the workspace is not trusted so extension is not loaded.

**Reply by rootsmusic** [reply]
Sorry @karthiknadig, I see Flake8, Pylint and Mypy in Output.  However, the command palette still doesn't have `Python: Select Linter`.  So I'm assuming that both Flake8 and Pylint are running, because their last output is "Received response".

I'm able to run a .py file, and Terminal shows no error.  But why does Mypy's Output show the following error?
> Received notification 'window/logMessage'.
> 4:1:4:1: error: Library stubs not installed for "pandas"  [import-untyped]
> 4:1:4:1: note: Hint: "python3 -m pip install pandas-stubs"
> note: (or run "mypy --install-types" to install all missing stub packages)
> note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports

**Reply by karthiknadig** [reply]
That is not really an error with your code. `mypy` needs additional libraries to better analyze your code, that is what it means. You can read more about it in the mypy documentation.

**Reply by karthiknadig** [reply, marked as answer]
You will not see Select linter. All linter extensions installed will automatically be active.
